tools/super-tox.py

```diff
@@ -415,7 +432,7 @@ def lxd_instance(
             instance.delete()
 
 
-@click.option("--workers", default=3, type=click.IntRange(1))
+@click.option("--workers", default=lambda: os.cpu_count() or 3, type=click.IntRange(1))
```

Comment:
Realistically, isn't this essentially always going to be `os.cpu_count()`? I know it's not guaranteed so could be None, but that seems unlikely anywhere this script would run. Probably the type checker wants some option here, but then I think the alternative should be 1 not 3.

In terms of what the default should be, I found cpu_count (20 on this device, some of which are 'performance' cores and some are not, and some are virtual cores) too many. If each tox was one process then it would probably be ok (although even then I think I would rather do cpu_count-1 or something like that to leave a bit of room for doing other work, assuming that tox would be maxing the ones it has). But several tests have tox using multiple processs (mysql-router is 120) so lots of workers ended up consuming way too much memory for me.

Maybe a default of cpu_count/2 for now? I would like to monkeypatch the tox configs to manage their concurrency levels so maybe it could be adjusted higher after that?

Reply:
Indeed this is outside of the tool's control.
`tox -e lint` is sequential and worker = 1.5 cpu cores seems to be a good starting point, because weirdly it's not entirely cpu-bound.
`tox` could start so many things, sequential or parallel.

I've also ran into a case where cpu was fine, but I was running out of memory.

I dunno if there is a default that's sensible for any or even most use cases.

Reply:
Would it make sense to guess the worker count based on `-e <thing>` flag, like cores for lint/static and 3 for unit tests?

Reply:
I think the only realistic option here is to default to `1` - `cpu_count()` feels like it will easily be too high in some situations, and everything else is just a magic number.

```suggestion
@click.option("--workers", default=1, type=click.IntRange(1))
```

In the future maybe we can have something that is env name based like you suggest, particularly if we are also tweaking what tox itself or pytest is doing.